recipes/cccl/all/test_package/conanfile.py

```diff
@@ -1,28 +1,27 @@
-from conans import ConanFile, tools
+from conan import ConanFile
+from conan.tools.build import can_run
+from conan.tools.env import Environment
 import os
 
 
 class CcclTestConan(ConanFile):
     settings = "os", "compiler", "build_type", "arch"
-    generators = "cmake"
-
+    generators = "VCVars"
+    
     @property
     def _settings_build(self):
         return getattr(self, "settings_build", self.settings)
 
     def build_requirements(self):
-        if self._settings_build.os == "Windows" and not tools.get_env("CONAN_BASH_PATH"):
+        if self._settings_build.os == "Windows" and not Environment().vars(self).get("CONAN_BASH_PATH"):
             self.build_requires("msys2/cci.latest")
 
     def build(self):
-        environment = {}
-        if self.settings.compiler == "Visual Studio":
-            environment.update(tools.vcvars_dict(self.settings))
-        with tools.environment_append(environment):
-            cxx = tools.get_env("CXX")
-            self.run("{cxx} {src} -o example".format(
-                cxx=cxx, src=os.path.join(self.source_folder, "example.cpp")), win_bash=self.settings.os is "Windows")
+        if can_run(self):
+            cxx = "sh cccl "
+            src = os.path.join(self.source_folder, "example.cpp")
+            self.run(f"{cxx} {src} -o example", win_bash=self.settings.os is "Windows", run_environment=True)
```

Comment:
not the conan v2 way. you should not use can_run here, but add ccl to tool_requires, change test_type to explicit, use VirtualBuildEnv, and use self.run in default scope (build scope). Moreover win_bash is enabled with self.win_bash = True